personhood/contract_popparty.go

```diff
@@ -0,0 +1,492 @@
+	if err != nil {
+		return
+	}
+	mr := make([]byte, 8)
+	binary.LittleEndian.PutUint64(mr, reward)
+	ctx := byzcoin.ClientTransaction{
+		Instructions: byzcoin.Instructions{byzcoin.Instruction{
+			InstanceID: byzcoin.NewInstanceID(dID),
+			Spawn: &byzcoin.Spawn{
+				ContractID: ContractPopPartyID,
+				Args: byzcoin.Arguments{{
+					Name:  "description",
+					Value: fsBuf,
+				}, {
+					Name:  "darcID",
+					Value: dID,
+				}, {
+					Name:  "miningReward",
+					Value: mr,
+				}},
+			},
+			SignerCounter: []uint64{signerCtrs.Counters[0] + 1},
+		}},
+	}
+	err = ctx.FillSignersAndSignWith(signers...)
+	if err != nil {
+		return
+	}
+	_, err = cl.AddTransactionAndWait(ctx, 5)
+	if err != nil {
+		return
+	}
+	return ctx.Instructions[0].DeriveID(""), nil
+}
+
+// PopPartyBarrier activates the barrier in the pop-party.
+func PopPartyBarrier(cl *byzcoin.Client, popIID byzcoin.InstanceID, signers ...darc.Signer) (err error) {
+	var sigStrs []string
+	for _, sig := range signers {
+		sigStrs = append(sigStrs, sig.Identity().String())
+	}
+	signerCtrs, err := cl.GetSignerCounters(sigStrs...)
+	if err != nil {
+		return
+	}
+
+	ctx := byzcoin.ClientTransaction{
+		Instructions: byzcoin.Instructions{byzcoin.Instruction{
+			InstanceID: popIID,
+			Invoke: &byzcoin.Invoke{
+				ContractID: ContractPopPartyID,
+				Command:    "barrier",
+			},
+			SignerCounter: []uint64{signerCtrs.Counters[0] + 1},
+		}},
+	}
+	err = ctx.FillSignersAndSignWith(signers...)
+	if err != nil {
+		return
+	}
+	_, err = cl.AddTransactionAndWait(ctx, 5)
+	return
+}
+
+// PopPartyFinalize sends the list of attendees to the party for finalization.
+func PopPartyFinalize(cl *byzcoin.Client, popIID byzcoin.InstanceID, atts Attendees, signers ...darc.Signer) (err error) {
+	var sigStrs []string
+	for _, sig := range signers {
+		sigStrs = append(sigStrs, sig.Identity().String())
+	}
+	signerCtrs, err := cl.GetSignerCounters(sigStrs...)
+	if err != nil {
+		return
+	}
+
+	attBuff, err := protobuf.Encode(&atts)
+	if err != nil {
+		return
+	}
+	ctx := byzcoin.ClientTransaction{
+		Instructions: byzcoin.Instructions{byzcoin.Instruction{
+			InstanceID: popIID,
+			Invoke: &byzcoin.Invoke{
+				ContractID: ContractPopPartyID,
+				Command:    "finalize",
+				Args: byzcoin.Arguments{
+					{
+						Name:  "attendees",
+						Value: attBuff,
+					},
+				},
+			},
+			SignerCounter: []uint64{signerCtrs.Counters[0] + 1},
+		}},
+	}
+	err = ctx.FillSignersAndSignWith(signers...)
+	if err != nil {
+		return
+	}
+	_, err = cl.AddTransactionAndWait(ctx, 5)
+	return
+}
+
+// PopPartyMine collects the reward. If 'atts' is nil, the party will be fetched from
+// byzcoin. One of 'coinIID' or 'd' must be set to either mine the coin, or spawn
+// a new darc and coin.
+func PopPartyMine(cl *byzcoin.Client, popIID byzcoin.InstanceID, kp key.Pair,
+	atts *Attendees, coinIID *byzcoin.InstanceID, d *darc.Darc) (err error) {
+	if atts == nil {
+		popProof, err := cl.GetProof(popIID.Slice())
```

Comment:
popProof is not checked here for being an inclusion proof of the requested key. (Or am I mis-remembering how to safely use GetProof?)

Reply:
added `Verify`.